### PR TITLE
GP2-1028 Target-market routing feature

### PR DESCRIPTION
**Pull request by bobmeredith**
- [x] Ticket exists in Jira  https://uktrade.atlassian.net/browse/GP2-1028 
 - [x] Jira ticket has the correct status.
 - [x] [Changelog](CHANGELOG.md) entry added.

 - [x] (if is a UI change) Includes screenshot(s) - ideally before and after, but at least after
 - [x] (if needs manual testing) How to test locally, including how to set up appropriate data
 
 - [X] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

**Comment by bobmeredith** [comment]
image

**Comment by bobmeredith** [comment]
To test - you will need to have the 'plan' and 'learn' routes set up in the dashboard in wagtail (and indeed have a dashboard page configured in wagtail)

- Add one or more markets to compare.  Note the 'next steps' block does not show.
- Select a market from the bottom panel - 'next-steps' appears.
- Remove the comparison market - 'next-steps' disappears.
- From the personalisation bar, check that the if a market that's in the comparison set it selected - 'next-steps' appears.
- Likewise, if a market is selected that is not in the comparison set, 'next-steps' disappears.

File: core/views.py [hunk 1]
```diff
@@ -124,10 +125,13 @@ def __init__(self):
     template_name = 'core/target_markets.html'
 
     def get_context_data(self, **kwargs):
+        dashboard_objects = DomesticDashboard.objects.all()
+        dashboard = dashboard_objects.count() and dashboard_objects[0]
```
Comment (stevejalim) [inline core/views.py:129]: Could do this as `DomesticDashboard.objects.live().first()` (which returns an object or None)

**Review by stevejalim — COMMENTED** [review]
Wagtail ~site~ side seems fine - minor suggestion to save a line of code

**Review by webbyfox — COMMENTED** [review]
Apart from Steve comment looks good

File: core/views.py (hunk 1 above)
Comment (bobmeredith) [reply core/views.py:129]: Brilliant! - I was wanting a better way to do this.  Thanks.

**Review by stevejalim — APPROVED** [review]
Backend LGTM

**Pull request merged**